### PR TITLE
[Security] Allow configuring a target url when switching user

### LINKED ISSUE #17021: [Security] Allow configuring a target url when switching user
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#46338
| PR author(s) | @94noni
| Merged in    | 6.2

### PR DESCRIPTION
**Pull request by 94noni**
Documents https://github.com/symfony/symfony/pull/47343
Supersed https://github.com/symfony/symfony/pull/46338 (feature changed)
Close https://github.com/symfony/symfony-docs/issues/17021

### CODE REVIEW

File: security/impersonating_user.rst [hunk 1]
```diff
@@ -244,6 +244,64 @@ also adjust the query parameter name via the ``parameter`` setting:
...
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/security
+                https://symfony.com/schema/dic/security/security-1.0.xsd">
+            <config>
+                <!-- ... -->
+
+                <firewall name="main">
+                    <!-- ... -->
+                    <switch-user target_url="/dasboard"/>
```
Comment (xabbuh) [inline security/impersonating_user.rst]:
```suggestion
                    <switch-user target-url="/dasboard"/>
```

**Comment by xabbuh** [comment]
looking at the example configuration I wonder if being able to configure a route name wouldn't be more sensible in most cases than having to configure the URL (path)

**Comment by 94noni** [comment]
> looking at the example configuration I wonder if being able to configure a route name wouldn't be more sensible in most cases than having to configure the URL (path)

I see, I've inspired myself with the [login default_target_path](https://symfony.com/doc/current/reference/configuration/security.html#default-target-path) iirc

swapping to a route name, what changes will be necessary on the code side? injecting the router that's all?

**Comment by xabbuh** [comment]
> swapping to a route name, what changes will be necessary on the code side? injecting the router that's all?

injecting a `UrlGeneratorInterface` should be enough, yes

**Comment by 94noni** [comment]
@xabbuh got it

I’ll try to handle the code change in the next days, and then rewrite this doc entry

Status: needs work

**Comment by 94noni** [comment]
> I’ll try to handle the code change in the next days, and then rewrite this doc entry

its finally done @xabbuh :)
status: needs review

many thanks

**Review by maxhelias — APPROVED** [review]

File: security/impersonating_user.rst [hunk 2]
```diff
@@ -244,6 +244,64 @@ also adjust the query parameter name via the ``parameter`` setting:
...
         };
 
+Redirecting to a specific target route
+--------------------------------------
+
+.. versionadded:: 6.2
+
+    The ``target_route`` configuration option was introduced in Symfony 6.2.
+
+This feature allows you to control the redirection targe route via ``target_route``.
```
Comment (xabbuh) [inline security/impersonating_user.rst]:
```suggestion
This feature allows you to control the redirection target route via ``target_route``.
```

File: security/impersonating_user.rst [hunk 3]
```diff
@@ -244,6 +244,64 @@ also adjust the query parameter name via the ``parameter`` setting:
             ;
         };
 
+Redirecting to a specific target route
```
Comment (xabbuh) [inline security/impersonating_user.rst]:
```suggestion
Redirecting to a Specific Target Route
```

File: security/impersonating_user.rst [hunk 4]
```diff
@@ -244,6 +244,64 @@ also adjust the query parameter name via the ``parameter`` setting:
...
+Redirecting to a Specific Target Route
+--------------------------------------
+
+.. versionadded:: 6.2
+
+    The ``target_route`` configuration option was introduced in Symfony 6.2.
+
+This feature allows you to control the redirection target route via ``target_route``.
+It works only in a statefull firewall.
+
```
Comment (OskarStark) [inline security/impersonating_user.rst]:
```suggestion
.. versionadded:: 6.2

    The ``target_route`` configuration option was introduced in Symfony 6.2.

.. note::

    It works only in a statefull firewall.

This feature allows you to control the redirection target route via ``target_route``.

```

Comment (94noni) [reply security/impersonating_user.rst]: Indeed the note is relevant

File: security/impersonating_user.rst [hunk 5]
```diff
@@ -244,6 +244,67 @@ also adjust the query parameter name via the ``parameter`` setting:
...
+Redirecting to a Specific Target Route
+--------------------------------------
+
+.. versionadded:: 6.2
+
+    The ``target_route`` configuration option was introduced in Symfony 6.2.
+
+.. note::
+
+    It works only in a statefull firewall.
```
Comment (xabbuh) [inline security/impersonating_user.rst]:
```suggestion
    It works only in a stateful firewall.
```

**Comment by 94noni** [comment]
I am afk so easier to accept suggested changes than using phone editor

**Review by xabbuh — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thanks Antoine.

**Pull request merged**